### DISCUSSION TITLE
scRNA snATAC Multiome10x

### DISCUSSION THREAD

**Discussion by matteozoia4** [GreenleafLab/ArchR#1583] [Questions / Documentation]
Dear all, 

I am new in ArchR and I would like to know whether the best approach to QCfilter RNA and ATAC experiments together is : 

a) QC ATACseq sample, plot the UMAP and integrate the RNAseq ?
b) QC on ATAC and RNA, plot the weighted UMAP ?

Many thanks for your help!
MZ

**Comment by rcorces** [comment]
It is unclear what you mean by "integration". Either way, my opinion is that one should always QC on both.

**Comment by matteozoia4** [comment]
Dear rcorces,

sorry for not being clear enough.
I have found two possible ways to integrate snATACseq and scRNAseq, and I was wondering about differences such as downsides between the two, which one would you advice to employ? 
 
1) Cross-platform linkage of scATAC-seq cells with scRNA-seq cells 
    (https://www.archrproject.com/bookdown/cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html)
2) Combine snATACseq with scRNAseq and meta.data files
     (https://greenleaflab.github.io/ArchR_2020/Ex-Analyze-Multiome.html)


Best regards and thank you for your time,

MZ

**Reply by rcorces** [reply]
> Cross-platform linkage of scATAC-seq cells with scRNA-seq cells
(https://www.archrproject.com/bookdown/cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html)

This is used when scATAC and scRNA data are acquired separately, not on the same exact cells

> Combine snATACseq with scRNAseq and meta.data files
(https://greenleaflab.github.io/ArchR_2020/Ex-Analyze-Multiome.html)

This is used for true multiomic data where scATAC and scRNA are acquired simultaneously in the same single cells using the 10x Multiome kit.

**Reply by matteozoia4** [reply]
Many thanks!
Regarding true multiome (my case) data I could combine the scATAC with an .rds object of an already filtered (QC Seurat) RNAseq?

Best regards and many thanks for your time!

**Reply by rcorces** [reply]
no the only supported use case currently is to import multiome RNA from CellRanger .h5 files. You can filter cells later using `subsetArchRProject()` if you'd like based on analyses in other tools.

**Reply by matteozoia4** [reply]
Many thanks!

**Reply by matteozoia4** [reply]
Dear R. Corces,

in case I have produced an ArchRProject (project) with 2 ATAC samples, how can I successfully add per sample the respective 10x multiome RNAseq features?

```
seRNA_E9.5 <- import10xFeatureMatrix(
  input = c("~/E9.5/9.5_filtered_feature_bc_matrix.h5"),
  names = c("seRNA_E9.5")

)seRNA_E10.5 <- import10xFeatureMatrix(
  input = c("~E10.5/10.5_filtered_feature_bc_matrix.h5"),
  names = c("seRNA_E10.5")
)

project <- addGeneExpressionMatrix(input = project, seRNA = seRNA_E9.5, force = TRUE)
project <- addGeneExpressionMatrix(input = project, seRNA = seRNA_E10.5, force = TRUE)
```

Have I to modify the **input = project** in a way that **Cell names in ArrowFiles match colnames in seRNA** ?

Best regards and thank you a lot!

**Reply by rcorces** [reply]
multiple h5 files should be passed to import10xFeatureMatrix. It will combine them.

**Reply by matteozoia4** [reply]
I see many thanks!

**Reply by matteozoia4** [reply]
Dear R. Corces,

Another question about redefining the filtering using some parameters such as: 
```
nCount_RNA < 25000 &
nCount_RNA > 500 &
percent_mito < 20 &
nFeature_RNA > 200 & 
nFeature_RNA < 7500 
```

How can I create the respective columns and then filter ?
```
pros <-proj[proj$TSSEnrichment > 6 & proj$nFrags > 2500  ......    proj$nCount_RNA < 25000 &     .....    !is.na(proj$Gex_nUMI)]
```

Ps: I haven't so far found a discussion or a script to do this, it is possible I miss it, sorry if the case!

Thank you in advance!
MZ

**Reply by rcorces** [reply]
**please please please** do not subset using `[]` but rather do this using `subsetArchRProject()`. I'm in the process of making this clear in the documentation.

create new columns in `cellColData` (for ex. with `addCellColData()` and then figure out which cells pass all of your criteria (for example using `intersect()` or similar).

**Reply by rcorces** [reply]
sorry - thats beyond the level of support I provide as it isnt directly related to ArchR.

**Comment by matteozoia4** [comment]
Dear Ryan,


I would like to plot linked gene expression with TFs on the UMAP embedding.


```

markerRNA <- getFeatures(myProj, select = paste(motifs, collapse="|"), useMatrix = "GeneIntegrationMatrix")

markerRNA

p <- plotEmbedding(

  ArchRProj = myProj,

  colorBy = "GeneIntegrationMatrix",

  name = sort(markerRNA),

  embedding = "IterativeLSI",

  continuousSet = "blueYellow",

  imputeWeights = getImputeWeights(myProj)

)

```


Using true Multiome-10x data I found myself in the situation where I don't possess a "groupRNA" parameter, this when trying to produce addGeneIntegrationMatrix().



```

myProj <- addGeneIntegrationMatrix(

  ArchRProj = myProj,

  useMatrix = "GeneScoreMatrix",

  matrixName = "GeneIntegrationMatrix",

  reducedDims = "IterativeLSI",

  seRNA = seRNA,

  groupATAC = "Clusters",

  groupRNA = "....",

  groupList = NULL,

  sampleCellsATAC = 10000,

  sampleCellsRNA = 10000,

  embeddingATAC = NULL,

  embeddingRNA = NULL,

  dimsToUse = 1:50,

  scaleDims = NULL,

  corCutOff = 0.75,

  plotUMAP = TRUE,

  UMAPParams = list(n_neighbors = 40, min_dist = 0.4, metric = "cosine", verbose =

                      FALSE),

  nGenes = 2000,

  useImputation = TRUE,

  reduction = "cca",

  addToArrow = TRUE,

  scaleTo = 10000,

  genesUse = NULL,

  nameCell = "predictedCell",

  nameGroup = "predictedGroup",

  nameScore = "predictedScore",

  transferParams = list(),

  threads = getArchRThreads(),

  verbose = TRUE,

  force = TRUE,

  logFile = createLogFile("addGeneIntegrationMatrix")

)

```


Best regards and thank you,


MZ


________________________________
Da: Ryan Corces ***@***.***>
Inviato: giovedì, 25. agosto 2022 15:26:16
A: GreenleafLab/ArchR
Cc: Zoia, Matteo (DBMR); Author
Oggetto: Re: [GreenleafLab/ArchR] scRNA snATAC integration : What is the best approach for QC and UMAPs production? (Discussion #1583)


Cross-platform linkage of scATAC-seq cells with scRNA-seq cells
(https://www.archrproject.com/bookdown/cross-platform-linkage-of-scatac-seq-cells-with-scrna-seq-cells.html)

This is used when scATAC and scRNA data are acquired separately, not on the same exact cells

Combine snATACseq with scRNAseq and meta.data files
(https://greenleaflab.github.io/ArchR_2020/Ex-Analyze-Multiome.html)

This is used for true multiomic data where scATAC and scRNA are acquired simultaneously in the same single cells using the 10x Multiome kit.
